core/src/com/unciv/logic/civilization/Civilization.kt

```diff
@@ -802,6 +810,80 @@ class Civilization : IsPartOfGameInfoSerialization {
     fun setAllyCiv(newAllyName: String?) { allyCivName = newAllyName }
 
     fun asPreview() = CivilizationInfoPreview(this)
+
+    fun calculateExploredRegion(position: Vector2, explorerPosition: Vector2?) {
```

Comment:
I hope after the existing comments I gave this will be more readable, but I still think this should definitely have more comments. You seem to be dealing with edge cases for world wrap that are unclear when reading the code, so just saying in lines like 'we passed over the edge of the world' could make things much clearer

Reply:
(also why do you need the explorer position?)

Reply:
ok, I will definitely add comments. I was wondering what you would say about this new feature in general, whether it deserves the right to life)

when exploring the last 2-4 columns of FoW tiles, it is difficult to determine from which side of the explored area to add them, because the new cell can be right in the middle between them. so I use the position of the explorer to determine from which side to expand the region